doc/stp/SONiC_PVST_HLD.md

```diff
@@ -0,0 +1,700 @@
+          * [STP Container](#stp-container)
+      * [SAI](#sai)
+      * [CLI](#cli)
+          * [Configuration Commands](#configuration-commands)
+          * [Show Commands](#show-commands)
+          * [Debug Commands](#debug-commands)
+          * [Clear Commands](#clear-commands)
+  * [Flow Diagrams](#flow-diagrams)
+  * [Serviceability and Debug](#serviceability-and-debug)
+  * [Warm Boot Support](#warm-boot-support)
+  * [Scalability](#scalability)
+  * [Unit Test](#unit-test)
+  
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author              | Change Description                |
+|:---:|:-----------:|:-------------------------:|-----------------------------------|
+| 0.1 | 05/02/2019  |     Sandeep, Praveen       | Initial version  |                
+| 0.2 | 05/02/2019  |     Sandeep, Praveen       | Incorporated Review comments  |                
+                                              
+
+# About this Manual
+This document provides general information about the PVST (Per VLAN spanning tree) feature implementation in SONiC.
+# Scope
+This document describes the high level design of PVST feature. 
+
+# Definition/Abbreviation
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| BPDU                     | Bridge protocol data unit           |
+| PVST                     | Per VLAN Spanning tree              |
+| STP                      | Spanning tree protocol              |
+
+
+# 1 Requirement Overview
+## 1.1 Functional Requirements
+
+
+ 1. Support PVST+ per VLAN spanning tree
+ 2. Support Portfast functionality
+ 3. Support Uplink fast functionality
+ 4. Support BPDU guard functionality
+ 5. Support Root guard functionality
+ 6. Bridge-id should include the VLAN id of the pvst instance along with bridge MAC address to derive a unique value for each instance.
+ 7. Port channel path cost will be same as the member port cost, it will not be cumulative of member ports cost
+ 8. DA MAC in case of PVST should be cisco pvst mac - 01:00:0C:CC:CC:CD
+ 9. Support protocol operation on static breakout ports
+ 10. Support protocol operation on Port-channel interfaces
+ 
+
+## 1.2 Configuration and Management Requirements
+This feature will support CLI and REST based configurations.
+ 1. Support CLI configurations as mentioned in section 3.6.2
+ 2. Support show commands as mentioned in section 3.6.3
+ 3. Support debug commands as mentioned in section 3.6.4
+ 4. Support Openconfig yang model - with extensions for supporting PVST 
+ 5. Support REST APIs for config and operational data
+
+## 1.3 Scalability Requirements
+16k port-vlan instances with max 255 STP instances.
+The scaling limit might differ depending on the platform and the CPU used, which needs to be determined based on testing.
+
+## 1.4 Warm Boot Requirements
+
+Warm boot will not be supported
```

Comment:
can you describe more on what exact do you mean by warm boot will not be supported?

Reply:
Warm reboot is not supported in this release. User is expected to do cold reboot when PVST is running so that topology will reconverge and traffic will be redirected via alternate paths.

If PVST is enabled and user tries to perform warm reboot a warning can be displayed indicating PVST doesn't support warm reboot.

Reply:
Document updated